src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs

```diff
@@ -50,6 +50,21 @@ public override void OnEnd(LogRecord data)
 
         if (activity?.IsAllDataRequested == true)
         {
+            try
+            {
+                if (this.options.Filter?.Invoke(data) == false)
+                {
+                    return;
+                }
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                OpenTelemetryExtensionsEventSource.Log.LogProcessorException($"Processing filter of category [{data.CategoryName}]", ex);
```

Comment:
nit: Let's add a dedicated event for this filter exception in `OpenTelemetryExtensionsEventSource`.

```csharp
    [NonEvent]
    public void LogRecordFilterException(string categoryName, Exception ex)
    {
        if (this.IsEnabled(EventLevel.Warning, (EventKeywords)(-1)))
        {
            this.LogRecordFilterException(categoryName, ex.ToInvariantString());
        }
    }

        [Event(2, Message = "Filter threw an exception, request will not be collected. CategoryName: '{0}', Exception: {1}.", Level = EventLevel.Warning)]
        public void LogRecordFilterException(string categoryName, string exception)
        {
            this.WriteEvent(2, categoryName, exception);
        }
```

Two reasons. I think a `Warning` is more appropriate. Debatable 😄 But also saves a string allocation when there isn't a listener.

Reply:
Done